sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -354,6 +351,10 @@ class Analyzer(override val catalogManager: CatalogManager) extends RuleExecutor
       UpdateAttributeNullability),
     Batch("UDF", Once,
       ResolveEncodersInUDF),
+    Batch("DML rewrite", fixedPoint,
```

Comment:
let's add a code comment to explain why it needs to be after `ResolveEncodersInUDF`

Reply:
Added some comments.

Reply:
Ya, thank you for adding this.